src/graphql/root/mutation/captcha-request-auth-code.ts

```diff
@@ -33,27 +33,27 @@ const CaptchaRequestAuthCodeMutation = GT.Field({
       }
     }
 
-    try {
-      // TODO: redo this with use-case and errors pattern
-      await verifyCaptchaAndReturnOTP({
-        phone,
-        geetest,
-        geetestChallenge,
-        geetestValidate,
-        geetestSeccode,
-        logger,
-        ip,
-      })
-      return {
-        errors: [],
-        success: true,
-      }
-    } catch (err) {
+    const result = await requestPhoneCodeWithCaptcha({
+      phone,
+      geetest,
+      geetestChallenge,
+      geetestValidate,
+      geetestSeccode,
+      logger,
+      ip,
+    })
+
+    if (result instanceof Error) {
       return {
```

Comment:
This is a little trick we've been doing in resolvers to convert domain errors to graphql errors

```suggestion
      const appErr = mapError(result)
      return {
```